### PR TITLE
Agent function http_route_in_use and tcp_port_in_use

### PR DESCRIPTION
**Pull request by stephdl**
http_route_in_use : Test if an url exists with the traefik api, return true is used or false

tcp_port_exists: Test if the port is used or not, return true is used or false

```
import agent
domain = agent.http_route_in_use(domain='rocky9-2.local', path='', expectation=404)
webpath = agent.http_route_in_use(domain='127.0.0.1', path='/foo', expectation=404)
port=agent.tcp_port_in_use(5555)
```

### CODE REVIEW

File: core/imageroot/usr/local/agent/pypkg/agent/__init__.py [hunk 1]
```diff
@@ -474,3 +479,60 @@ def get_smarthost_settings(rdb):
...
+        urllib.request.urlopen(req, context=ctx)
+    except urllib.error.HTTPError as d:
+        https = d.code
+
+    if http == expectation and  https == expectation:
+        # there is no website on http or https -> OK
+        return False
+    else:
+        # path exists -> nok
+        return True
```
Comment (DavidePrincipi) [inline core/imageroot/usr/local/agent/pypkg/agent/__init__.py:520]: That's all...

```suggestion
def http_route_in_use(domain=None, path='', expectation=404):
    """ we search if the domain (foo.com) or the webpath (/foo) is used or not by a service
        return true if the route is used, false is the route is not used"""
    try:
        req = urllib.request.Request('http://127.0.0.1'+path)
        if domain:
            req.add_header('Host', domain)
        urllib.request.urlopen(req, context=ctx)
    except urllib.error.HTTPError as e:
        if e.code == expectation:
            # there is no website on http or https -> OK
            return False

    return True
```

Comment (DavidePrincipi) [reply core/imageroot/usr/local/agent/pypkg/agent/__init__.py:520]: ...I guess the urlopen() function could also succeed without raising any exception. What happens with status code 200?

Comment (stephdl) [reply core/imageroot/usr/local/agent/pypkg/agent/__init__.py:520]: yes only 404 is the return that the route does not exist

Comment (DavidePrincipi) [reply core/imageroot/usr/local/agent/pypkg/agent/__init__.py:520]: Edit: added `method="HEAD"` to avoid fetching page contents

**Review by DavidePrincipi — APPROVED** [review]

File: core/imageroot/usr/local/agent/pypkg/agent/__init__.py [hunk 2]
```diff
@@ -32,6 +32,11 @@
 import shlex
 import warnings
 import agent.tasks
+import socket
+import errno
+import urllib.request
+import urllib.error
+import ssl
```
Comment (DavidePrincipi) [inline core/imageroot/usr/local/agent/pypkg/agent/__init__.py:39]:
```suggestion
```

**Pull request merged**